/*
 * GroupMessageMgr.h
 *
 *  Created on: 2014年5月22日
 *      Author: root
 */

#ifndef GROUPMESSAGEMGR_H_
#define GROUPMESSAGEMGR_H_

#include "Network/MessageHandler.h"
#include "MsgDefineMacro.h"
#include "ServerMsgDefine.h"
#include <map>
#include "../Object/Player/Player.h"
#include "../QuickMessage.h"

using namespace std;
using namespace CommBaseOut;

class GroupMessageMgr : public Singleton<GroupMessageMgr>
{
public:
	GroupMessageMgr();
public:
	~GroupMessageMgr();

	DEF_MSG_QUICK_SIMPLE_DECLARE_FUN_H(MSG_REQ_C2GM_CREATETEAM);		//请求创建队伍
	DEF_MSG_QUICK_REQUEST_DECLARE_FUN_H(MSG_REQ_C2GM_INVITETOTEAM);	//邀请入队
	DEF_MSG_QUICK_REQUEST_DECLARE_FUN_H(MSG_REQ_C2GM_REPLYINVITE);		//邀请回应
	DEF_MSG_QUICK_REQUEST_DECLARE_FUN_H(MSG_REQ_C2GM_KICKMEMBER);		//踢出队友
	DEF_MSG_QUICK_REQUEST_DECLARE_FUN_H(MSG_REQ_C2GM_CHANGELEADER);		//更换队长
	DEF_MSG_QUICK_REQUEST_DECLARE_FUN_H(MSG_REQ_C2GM_APPLICATETOTEAM);		//申请入队
	DEF_MSG_QUICK_REQUEST_DECLARE_FUN_H(MSG_REQ_C2GM_GETTEAMID);		//请求附近的队伍ID
	DEF_MSG_QUICK_REQUEST_DECLARE_FUN_H(MSG_REQ_C2GM_GETPLAYERS);		//请求附近玩家
	DEF_MSG_QUICK_REQUEST_DECLARE_FUN_H(MSG_REQ_C2GM_LEAVETEAM);		//离开队伍
	DEF_MSG_QUICK_REQUEST_DECLARE_FUN_H(MSG_REQ_C2GM_LEADERRETAPP);		//队长回复申请者
	DEF_MSG_QUICK_REQUEST_DECLARE_FUN_H(MSG_REQ_C2GM_LOOKFORTEAMINFO);		//查看队伍信息
	DEF_MSG_QUICK_REQUEST_DECLARE_FUN_H(MSG_REQ_C2GM_UPDATETEAMINFO);		//更新队伍信息
	DEF_MSG_QUICK_REQUEST_DECLARE_FUN_H(MSG_REQ_SYSTEAMMEMBERMAP);		//同步队员位置信息

	DEF_MSG_QUICK_REQUEST_DECLARE_FUN_H(MSG_REQ_C2GM_FUZZYSELECTPLAYERINFO);		//模糊查询 在线玩家

	DEF_MSG_QUICK_REQUEST_DECLARE_FUN_H(MSG_REQ_MEMBERAPPLYTOINSTANCE);		//队长创建副本，队员是否愿意进入询问


};

#endif /* GROUPMESSAGE_H_ */
